### PR TITLE
[DevOps] Rename the yaml-templates resources to match the MAUI repo.

### PR DESCRIPTION
**Pull request by mandel-macaque**
In an attempt to unify the build of all the SDKs we need to be rename all shared resources between the CIs to all use the same name.

This commit renames the resource to use the pattern used by MAUI.

Partial fix for https://github.com/xamarin/sdk-insertions/issues/39

### CODE REVIEW

**Review by rolfbjarne — APPROVED** [review]

File: tools/devops/automation/build-cronjob.yml [hunk 1]
```diff
@@ -132,7 +132,7 @@ resources:
     checkoutOptions:
       submodules: true
 
-  - repository: templates
+  - repository: xamarin-templates
```
Comment (pjcollins) [inline tools/devops/automation/build-cronjob.yml:135]: Maybe it would be best to use the repository name `yaml-templates` here instead?

Comment (mandel-macaque) [reply tools/devops/automation/build-cronjob.yml:135]: I'm using the maui name, we will probably want to fix this in the future, but is a good starting point.

**Review by pjcollins — APPROVED** [review]

**Pull request merged**